test/src/test/java/hudson/slaves/DummyCloudImpl.java

```diff
@@ -64,12 +69,26 @@
      */
     public Label label;
 
+    public List<NodeProperty<?>> getNodeProperties() {
+        return this.nodeProperties;
+    }
+
+    List<NodeProperty<?>> nodeProperties =
+            new ArrayList<NodeProperty<?>>();
+
     public DummyCloudImpl(JenkinsRule rule, int delay) {
         super("test");
         this.rule = rule;
         this.delay = delay;
     }
 
```

Comment:
NIT: better to have only one constructor with logic and to pass <code>this(rule, delay, new ArrayList<NodeProperty<?>>())</code> in the second. It ay be also better if you use a getter/setter to specify the data in order to follow the syntax of other implementations